base/dict.jl

```diff
@@ -778,14 +778,35 @@ end
 """
     ImmutableDict
 
-`ImmutableDict` is a dictionary implemented as an immutable linked list,
-which is optimal for small dictionaries that are constructed over many individual insertions.
-Note that it is not possible to remove a value, although it can be partially overridden and hidden
-by inserting a new value with the same key.
+`ImmutableDict` is a dictionary implemented as an immutable linked
+list. It's for small heavily-used mappings with a few entries, where
+the overhead of accessing a hash table is higher than that of a linear
+search of a linked list.  `ImmutableDict` is `public` rather than
+`export`; to access it one must write `Base.ImmutableDict`
```

Comment:
This can be conveyed by example by using `Base.ImmutableDict` in the opening line
```suggestion
search of a linked list.
```